Common/interface/CallbackWrapper.hpp

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright 2019-2022 Diligent Graphics LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  In no event and under no legal theory, whether in tort (including negligence),
+ *  contract, or otherwise, unless required by applicable law (such as deliberate
+ *  and grossly negligent acts) or agreed to in writing, shall any Contributor be
+ *  liable for any damages, including any direct, indirect, special, incidental,
+ *  or consequential damages of any character arising as a result of this License or
+ *  out of the use or inability to use the software (including but not limited to damages
+ *  for loss of goodwill, work stoppage, computer failure or malfunction, or any and
+ *  all other commercial damages or losses), even if such Contributor has been advised
+ *  of the possibility of such damages.
+ */
+
+#include <tuple>
+
+namespace Diligent
+{
+
+template <typename T>
+struct FunctionTraits : public FunctionTraits<decltype(&T::operator())>
+{};
+
+template <typename ClassType, typename ReturnType, typename... Args>
+struct FunctionTraits<ReturnType (ClassType::*)(Args...) const>
+{
+    using ResultType = ReturnType;
+
+    template <std::size_t Index>
+    using Argument = typename std::tuple_element<Index, std::tuple<Args...>>::type;
+
+    static const std::size_t Arity = sizeof...(Args);
+};
+
+template <typename Lambda, typename Arg>
+struct Callback
+{
+    void (*Function)(Arg, void*);
+    Lambda State;
+};
+
+template <typename Lambda, typename Arg = typename FunctionTraits<Lambda>::template Argument<0>>
+Callback<Lambda, Arg> WrapCallback(Lambda&& L)
+{
+    static_assert(FunctionTraits<Lambda>::Arity == 1, "Check number of arguments");
```

Comment:
There is a bug here: when using this as LoadPipelineStateInfo::Modify, the Arity is 2